app/forms/validators.py

```diff
@@ -49,15 +54,15 @@ def __call__(
         field: Union[DecimalFieldWithSeparator, IntegerFieldWithSeparator],
     ) -> None:
         try:
-            Decimal(
-                field.raw_data[0].replace(
-                    numbers.get_group_symbol(flask_babel.get_locale()), ""
-                )
-            )
+            Decimal(sanitise_number(number=field.raw_data[0]))
         except (ValueError, TypeError, InvalidOperation, AttributeError) as exc:
             raise validators.StopValidation(self.message) from exc
 
-        if "e" in field.raw_data[0].lower():
+        if "e" in field.raw_data[0].lower() or math.isnan(
+            Decimal(
+                sanitise_number(number=field.raw_data[0])
+            )  # number is sanitised to guard against inputs like `,NaN_` etc
+        ):
```

Comment:
Minor, this can be simplified, since you do the same call already.

```py
        try:
            # number is sanitised to guard against inputs like `,NaN_` etc
            number = Decimal(sanitise_number(number=field.raw_data[0]))
        except (ValueError, TypeError, InvalidOperation, AttributeError) as exc:
            raise validators.StopValidation(self.message) from exc

        if "e" in field.raw_data[0].lower() or math.isnan(number):
            raise validators.StopValidation(self.message)
```

Reply:
Yeah, I was suspecting some potential refactor here but couldn't spot it like you. Makes sense, no point calling it twice.